_posts/2021-06-10-solidity-0.8.5-release-announcement.md

```diff
@@ -0,0 +1,162 @@
+---
+layout: post
+published: true
+title: 'Solidity 0.8.5 Release Announcement'
+date: '2021-06-10'
+author: Solidity Team
+category: Releases
+---
+
+[Solidity v0.8.5](https://github.com/ethereum/solidity/releases/tag/v0.8.5) allows conversions
+from ``bytes`` to ``bytesNN`` values, adds the ``verbatim``builtin function to inject
+arbitrary bytecode in Yul and fixes several smaller bugs.
+
+
+## Notable New Features
+### Bytes Conversion
+
+This release introduces the ability to convert ``bytes`` and ``bytes`` slices to fixed bytes types ``bytes1`` / ... / ``bytes32``.
+In case a byte array is longer than the target fixed bytes type, it will be truncated at the end:
+
+```solidity
+function f(bytes calldata c) public view returns (bytes8) {
+	require(c.length == 16, "");
+	bytes8 b = bytes8(c); // c is longer than 8 bytes, so truncation happens
+	return b;
+}
+```
+
+If the array is shorter than the target fixed type, it will be padded with zeros at the end:
+
+```solidity
+function f(bytes calldata c) public view returns (bytes8) {
+	require(c.length == 30, "");
+	bytes32 b = bytes32(c); // c is 30 bytes long, so b will be padded with two zero bytes at the end
+	return b;
+}
+```
+
+A nice example of using the `bytes` conversion feature would be its application in proxies:
+
+```solidity
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >0.8.4 <0.9.0;
+contract Proxy {
+	/// @dev Address of the client contract managed by proxy i.e., this contract
+	address client;
+	constructor(address _client) {
+		client = _client;
+	}
+	/// Forward call to "setOwner(address)" that is implemented by client
+	/// after doing basic validation on the address argument.
+	function forward(bytes calldata _payload) external {
+		// Since ABI decoding requires padded data, we cannot
+		// use abi.decode(_payload[:4], (bytes4)).
+		bytes4 sig =
```

Comment:
Maybe put this in comments and just briefly say "this was the way this had to be done before"? Or even put it after or before contract inside the text?

Reply:
Makes sense, will do it now